tasks/create-internal-request/create-internal-request.yaml

```diff
@@ -53,6 +53,46 @@ spec:
     - name: input
       description: Workspace to store the params and responses for the internalRequest
   steps:
+    - name: merge-internal-request-parameters
+      image:
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      script: |
+          #!/usr/bin/env bash
+          DATA_FILE="$(workspaces.input.path)/$(params.dataPath)"
+          if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            exit 1
+          fi
+
+          IR_PARAMS_FILE="$(workspaces.input.path)/$(params.subdirectory)/ir-params-$$.txt"
+          cat > ${IR_PARAMS_FILE} <<JSON
+          $(params.internalRequestParams)
+          JSON
+
+          # to keep the compatibility the internalrequest parameters must be merged
+          # with the data json
+          JSON_PARAMS="$(jq -c '.' ${IR_PARAMS_FILE})"
+          INDEX=$(jq '.|length' ${IR_PARAMS_FILE})
+          jq '.fbc' $(workspaces.input.path)/$(params.dataPath) |yq -oy -P |while read PARAM_VALUE; do
+            read -r PARAM VALUE <<< "${PARAM_VALUE}"
+
+            PARAM="${PARAM%*:}"
+            # parameters from the task should replace the ones from the data.json
+            if [[ -n $(jq ".[] | select(.name==\"${PARAM}\")" $IR_PARAMS_FILE) ]]; then
+              continue
+            fi
+
+            # xargs is used to remove the extra ""
+            JSON_PARAMS=$(jq ".[${i}] += {name: \"${PARAM:}\", value: \"$(echo ${VALUE}| xargs)\"}" \
```

Comment:
Maybe I'm blind, but where does the `${i}` come from? Also, why is the `xargs` needed? Can you give an example? I'm pretty sure we can avoid it.

Reply:
The value is set in the RPA as `"80"` for instance. So when it gets here, without the xargs it becomes `""80""`. I appreciate if you have a better suggestion. Just seemed quick to do this.

Reply:
Indeed. It was supposed to be `${INDEX}` not `${i}`

Reply:
> So when it gets here, without the xargs it becomes ""80""

I don't see a reason why that should be the case. Maybe it has something to do with the funky parsing via `tsv` format and `read`? I think if you simply get the value directly via `jq` as I suggest in another comment, it should be ok. And if not, let me know and I will test it.

Reply:
Ok. removed as well. It seems without the `@tsv` it does not need it.

Reply:
Cool.